### PR TITLE
[stdlib] De-gyb Arrays.swift.gyb

### PR DESCRIPTION
**Pull request by lancep**
Split up Array, ArraySlice and ContiguousArray into one file each, with an ArrayShared file for common stuff.

### CODE REVIEW

File: stdlib/public/core/ArrayShared.swift [hunk 1]
```diff
@@ -0,0 +1,375 @@
+//===--- Arrays.swift.gyb -------------------------------------*- swift -*-===//
```
Comment (gottesmm) [inline stdlib/public/core/ArrayShared.swift]: Wrong file name.

File: stdlib/public/core/ArraysCommon.swift [hunk 2]
```diff
@@ -0,0 +1,407 @@
+// ###sourceLocation(file: "/Users/lanceparker/Development/swift-source/swift/stdlib/public/core/Arrays.swift.gyb", line: 1)
```
Comment (gottesmm) [inline stdlib/public/core/ArraysCommon.swift]: Bad title and ### source location. You should probably grep for these.

**Review by gottesmm — CHANGES_REQUESTED** [review]
+1 on less gyb! Found some nits.

**Review by gottesmm — APPROVED** [review]

**Comment by gottesmm** [comment]
(sorry I just meant to leave comments, not approve/disapprove)

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test

**Pull request merged**